models/staging/stg_nhl__shifts.sql

```diff
@@ -0,0 +1,242 @@
+        and not (s.gameid = 2020020745 and s.eventnumber = 485 and s.id = 10875539) -- time of goal was off
+        and not (s.gameid = 2020020767 and s.eventnumber = 209 and s.id = 10639438) -- time of goal was off
+)
+
+, shifts_time as (
+    select
+        case
+            when period = 1
+                then (shifts_raw.start_time_mins * 60) + (shifts_raw.start_time_seconds)
+            when period = 2
+                then (20 * 60) + ((shifts_raw.start_time_mins * 60) + (shifts_raw.start_time_seconds))
+            when period = 3
+                then (40 * 60) + ((shifts_raw.start_time_mins * 60) + (shifts_raw.start_time_seconds))
+            when period = 4
+                then (60 * 60) + ((shifts_raw.start_time_mins * 60) + (shifts_raw.start_time_seconds))
+            when period = 5
+                then (80 * 60) + ((shifts_raw.start_time_mins * 60) + (shifts_raw.start_time_seconds))
+            when period = 6
+                then (100 * 60) + ((shifts_raw.start_time_mins * 60) + (shifts_raw.start_time_seconds))
+            when period = 7
+                then (120 * 60) + ((shifts_raw.start_time_mins * 60) + (shifts_raw.start_time_seconds))
+            when period = 8
+                then (140 * 60) + ((shifts_raw.start_time_mins * 60) + (shifts_raw.start_time_seconds))
+            when period = 9
+                then (160 * 60) + ((shifts_raw.start_time_mins * 60) + (shifts_raw.start_time_seconds))
+            when period = 10
+                then (180 * 60) + ((shifts_raw.start_time_mins * 60) + (shifts_raw.start_time_seconds))
+        end as start_seconds_elapsed
+        , case
+            when shifts_raw.type_code = 505
+                then 0
+            else (shifts_raw.duration_mins * 60) + (shifts_raw.duration_seconds)
+        end as duration_seconds_elapsed
+        , case
+            when shifts_raw.goal_assist_count = '2 assisters'
+                 then trim(split(shifts_raw.goal_assisters, ',')[offset(0)])
+            when shifts_raw.goal_assist_count = '1 assister'
+                 then shifts_raw.goal_assisters
+        end as goal_primary_assister_full_name
+        , case
+            when shifts_raw.goal_assist_count = '2 assisters'
+                 then trim(split(shifts_raw.goal_assisters, ', ')[offset(1)])
+        end as goal_secondary_assister_full_name
+        , shifts_raw.*
+    from shifts_raw
+    where 1 = 1
+        and shifts_raw.period_type <> 'shootout'
+        and shifts_raw.end_time <> '' -- remove 1129 shifts, dups
+
+)
+
+, new_shift_id as (
+    select
+        case
+            when is_goal is true
+                then concat('_gameid_', game_id, '_playerid_', player_id, '_starsecondselapsed_', start_seconds_elapsed, '_isgoal_true')
+            when is_goal is false
+                then concat('_gameid_', game_id, '_playerid_', player_id, '_starsecondselapsed_', start_seconds_elapsed, '_duration_', duration_seconds_elapsed)
+        end as new_shift_id
```

Comment:
good shout, will rename it. I think it might make sense to convert it to a JSON construct at some point since the logic serves me for now to a quick debug